# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/Notification.h
# Commit: 1f3365d25033
# Full Hash: 1f3365d25033f9dbc52b13e8fc4c64f3cdd35946
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-12 09:32:50
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 2: Create PNotification via PBackground r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227687
# ==============================================================================

diff -r cc25e5c94813 -r 1f3365d25033 dom/notification/Notification.h
--- a/dom/notification/Notification.h	Mon Nov 11 20:34:11 2024 +0000
+++ b/dom/notification/Notification.h	Mon Nov 11 20:34:11 2024 +0000
@@ -11,6 +11,7 @@
 #include "mozilla/GlobalFreezeObserver.h"
 #include "mozilla/UniquePtr.h"
 #include "mozilla/dom/NotificationBinding.h"
+#include "mozilla/dom/notification/NotificationChild.h"
 #include "mozilla/dom/WorkerPrivate.h"
 #include "mozilla/dom/quota/QuotaCommon.h"
 
@@ -273,6 +274,8 @@
     mScope = aScope;
   }
 
+  WeakPtr<notification::NotificationChild> mActor;
+
   const nsString mID;
   const nsString mTitle;
   const nsString mBody;
@@ -318,6 +321,8 @@
       const NotificationOptions& aOptions, const nsAString& aScope,
       ErrorResult& aRv);
 
+  bool CreateActor(Promise* aPromise);
+
   nsIPrincipal* GetPrincipal();
 
   nsresult PersistNotification();